### PR TITLE
(RFC) agent multi inputs

### PR DESCRIPTION
**Pull request by hwchase17**
basic idea: all agent prompts need to end with thoughts input variable (where the inner monologue of the agent occurs)

but other than that, can take arbitary keys

### CODE REVIEW

File: langchain/agents/agent.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ def input_keys(self) -> List[str]:
 
         :meta private:
         """
-        return [self.input_key]
+        return set(self.llm_chain.input_keys) - {"thoughts"}
```
Comment (JohnNay) [inline langchain/agents/agent.py:40]: Where is this `{"thoughts"}` coming from into this function?

**Review by JohnNay — COMMENTED** [review]
I get that it is bringing the `thoughts` along the way as the agent moves forward iterates through LLM calls, but I'm not quite understanding the how this lets us specify more than just an `input` as the initial input, i.e. how we could do like we can with chains and specify arbitrary num of args to a call that provide a bunch of text inputs?

**Review by JohnNay — APPROVED** [review]
lgtm - i tested it too

**Comment by JohnNay** [comment]
anything left to do before merging this? it worked for me.

**Comment by hwchase17** [comment]
> anything left to do before merging this? it worked for me.

just want to pair this up with some other agent changes/refactors... should be out monday at the latest. sorry for the delay - things have been a bit hectic

**Comment by JohnNay** [comment]
no worries at all. was jw if there was anything I could have helped with on it

**Comment by hwchase17** [comment]
closing because this got merged into the big agent refactor PR

**Pull request closed without merging**